### DISCUSSION TITLE
2006/11/23/%E4%B8%80%E4%B8%AA-xcontrol-%E7%9A%84%E5%AE%9E%E4%BE%8B

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#131] [Announcements]
# 2006/11/23/%E4%B8%80%E4%B8%AA-xcontrol-%E7%9A%84%E5%AE%9E%E4%BE%8B

XControl 与 .ctl 用户定义控件相比，其最大的提高就在于它不但可以定义控件的外观，还可以定义控件的行为。 在 XControl 出现之前，同样可以在程序中编写代码，控制程序的行为。在《用 XControl 实现面向组件的编程》一文中提到了，这种方法在程序模块划分上有缺陷。如果用户想发布一个带有特定行为的控件也是不可能的，因为控制控件行为的代码，是同其它代码混杂在一起的。

https://blog.qizhen.xyz/2006/11/23/%E4%B8%80%E4%B8%AA-xcontrol-%E7%9A%84%E5%AE%9E%E4%BE%8B

**Comment by ruanqizhen** [comment]
之前博客下的留言：


Qizhen
2009年07月6日 15:33 编辑
XControl有大量额外代码，效率肯定会受影响。2ms应当足以完成采集、显示工作了。但是这样做意义不大，为了提高效率应当每200ms采集100数据，而不是每2ms采集一个数据。

回复
gufei
2009年07月5日 21:38 编辑
阮老师，请问使用XControl会不会影响效率，我发现把代码直接写到vi中，比写到XControl中效率高很多啊、、、还有一个问题，我现在有个大数组，想模拟实时采集、处理显示，采样频率是500hz，即每2ms读取数组的一列，我的做法是用个循环，每2ms循环一次，处理和显示放在循环中，发现2ms内可能会处理不完操作，那就可能放弃操作了，请问有什么办法实现我的要求？

回复
gufei
2009年01月27日 10:59 编辑
提个建议啊，能不能给你的例子一些详细的解释：）感觉例子就没有什么解释啊，谢谢

回复
刘伟印
2008年09月5日 11:52 编辑
不错，支持一下

回复
Qizhen
2007年08月31日 13:29 编辑
最近我放这些程序的 google page 被政府和谐了。等我有时间换个地方放这些示例。

回复
Unknown
2007年08月31日 11:08 编辑
范例程序没有一个能下载的！！ 

回复
Zhang Ping
2007年07月25日 10:17 编辑
仰望一下 o(∩_∩)o… 

回复
Unknown
2007年03月8日 13:55 编辑
受教了,如果多个XControl实例同时运行能准确无误就太好了.

回复
Wu
2006年11月23日 22:16 编辑
高！
